internal/xds/translator/proxy_protocol.go

```diff
@@ -10,35 +10,46 @@ import (
 	proxyprotocolv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/listener/proxy_protocol/v3"
 	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
 	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/envoyproxy/gateway/internal/ir"
 )
 
 // patchProxyProtocolFilter builds and appends the Proxy Protocol Filter to the
 // HTTP Listener's Listener Filters if applicable.
-func patchProxyProtocolFilter(xdsListener *listenerv3.Listener, enableProxyProtocol bool) {
-	// Return early if unset
-	if xdsListener == nil || !enableProxyProtocol {
+func patchProxyProtocolFilter(xdsListener *listenerv3.Listener, enableProxyProtocol bool, proxyProtocolSettings *ir.ProxyProtocolSettings) {
```

Comment:
can `irListener.ProxyProtocol != nil` be moved into this func to reduce the args In the API signature ?